content/en/docs/Configuration/p8s-jaeger-grafana/prometheus.md

```diff
@@ -0,0 +1,183 @@
+can't reach it, Kiali won't work properly.
+
+By default, Kiali assumes that Prometheus is available at the URL of the form
+`http://prometheus.<istio_namespace_name>:9090`, which is the usual case if you
+are using [the Prometheus Istio
+add-on](https://istio.io/latest/docs/ops/integrations/prometheus/#option-1-quick-start).
+If your Prometheus instance has a different service name or is installed to a
+different namespace, you must manually provide the endpoint where it is
+available, like in the following example:
+
+```yaml
+spec:
+  external_services:
+    prometheus:
+      # Prometheus service name is "metrics" and is in the "telemetry" namespace
+      url: "http://metrics.telemetry:9090/"
+```
+
+{{% alert color="success" %}}
+Notice that you don't need to expose Prometheus outside the cluster. It is
+enough to provide the Kubernetes internal service URL.
+{{% /alert %}}
+
+Kiali maintains an internal cache of some Prometheus queries to improve
+performance (mainly, the queries to calculate Health indicators). It
+would be very rare to see data delays, but should you notice any delays you may
+tune caching parameters to values that work better for your environment. These
+are the default values:
+
+```yaml
+spec:
+  external_services:
+    prometheus:
+      cache_enabled: true
+      # Per-query expiration in seconds
+      cache_duration: 10
+      # Global cache expiration in seconds. Think of it as
+      # the "reset" or "garbage collection" interval.
+      cache_expiration: 300
+```
+
+### Compatibility with Prometheus-like servers
+
+Although Kiali assumes a Prometheus server and is tested against it, there are
+<abbr title="Time series databases">TSDBs</abbr> that can be used as Prometheus
+replacement despite not implementing the full Prometheus API. 
+
+Community users have faced two issues when using Prometheus-like TSDBs:
+* Kiali may report that the TSDB is unreachable, and/or
+* Kiali may show empty metrics if the TSBD does not implement the `/api/v1/status/config`.
+
+To fix these issues, you may need to provide a custom health check endpoint for
+the TSDB and/or manually provide the configurations that Kiali reads from the
+`/api/v1/status/config` API endpoint:
+
+```yaml
+spec:
+  external_services:
+    prometheus:
+      # Fix the "Unreachable" metrics server warning.
+      health_check_url: "http://custom-tsdb-health-check-url"
+      # Fix for the empty metrics dashboards
+      thanos_proxy:
+        enabled: true
+        retention_period: "7d"
+        scrape_interval: "30s"
+```
+
+## Prometheus Tuning
+
+Production environments should not be using the Istio Prometheus add-on, or carrying over its configuration settings.  That is useful only for small, or demo installations.  Instead, Prometheus should have been installed in a production-oriented way, according to [Prometheus documentation](https://prometheus.io/docs/prometheus/latest/installation).
```

Comment:
This will be the first of many suggestions to prune superflouous commas I see in here.

```suggestion
Production environments should not be using the Istio Prometheus add-on or carrying over its configuration settings.  That is useful only for small or demo installations.  Instead, Prometheus should have been installed in a production-oriented way following the recommendations in the [Prometheus documentation](https://prometheus.io/docs/prometheus/latest/installation).
```